Makefile

```diff
@@ -3,6 +3,9 @@ all: build
 fmt:
 	go fmt ./...
 
+vet:
+	go vet ./pkg/... ./cmd/...
```

Comment:
If we move `capi/` into `pkg/` this will no longer work:

```
# github.com/virtblocks/virtblocks/pkg/capi
capi.c:1:10: fatal error: libvirtblocks.h: No such file or directory
    1 | #include "libvirtblocks.h"
      |          ^~~~~~~~~~~~~~~~~
compilation terminated
```

That's alright, we're just gonna have to make the Makefile a bit more clever - I haven't found a way to tell `go vet` it should be ignore specific folders.

Reply:
Yes, I ran into that too. Another option would be to always check-in the header file. In general it is very common to check in generated sources in golang to allow building and verification. However, since we definitely don't want to check in the library, it is not really helpful in this case. :/

Reply:
> Yes, I ran into that too. Another option would be to always check-in the header file. In general it is very common to check in generated sources in golang to allow building and verification. However, since we definitely don't want to check in the library, it is not really helpful in this case. :/

Checking it the library would not be necessary - the header would be enough. But I'm really not a fan of checking in generated source files, so I'd rather avoid that.

Another option would be to make `build` a dependency of `vet`, so by the time we run `go vet` we have already generated the header file... Would that make sense?

Just making the Makefile smart enough to skip `capi/` is still a perfectly valid option, of course :)

Reply:
> Another option would be to make `build` a dependency of `vet`, so by the time we run `go vet` we have already generated the header file... Would that make sense?
> 
> Just making the Makefile smart enough to skip `capi/` is still a perfectly valid option, of course :)

Let's have the `build` dependency. I find it good if `vet` can validate all code without artificial excludes.